### PR TITLE
Native extension

### PR DESCRIPTION
**Pull request by renam**
Adds objects to encode a Native ad request as part of n extension to the Native object

### CODE REVIEW

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        h = format.h,
+        w = format.w
+    )
+)
+
+public fun Asset.Companion.video(
+    id: Int,
+    required: Boolean,
+    mimes: Array<String>? = null,
+    minduration: Int = 0,
```
Comment (standa-dev) [inline kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: I think we agreed to set the default minduration = 1

**Review by standa-dev — APPROVED** [review]

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt (hunk 1 above)
Comment (renam) [reply kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: updated

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    length: Int
+): Asset = Asset(
+        id,
+        required.apiValue,
+        title = Asset.TitleObject(
+            length = length
+        )
+    )
+
+public fun Asset.Companion.image(
```
Comment (JasonSznol) [inline kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: These methods should be moved into the request module and should include code documentation, especially around the ID field which requires each asset id be unique.

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        @JvmField @SerialName("protocols") public var protocols: ByteArray? = null,
+    )
+    @Serializable
+    public class DataObject(
+        @JvmField @SerialName("type") public var type: Byte,
+        @JvmField @SerialName("len") public var len: Int,
+    )
+}
+
+private inline val Boolean.apiValue : Byte
```
Comment (JasonSznol) [inline kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: This method already exists in the request module. 

From a style perspective, if this were to remain here it should be at the bottom of the file as its the least relevant to a consumer given its access level.

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt (hunk 2 above)
Comment (renam) [reply kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: moved out of here

File: kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt (hunk 3 above)
Comment (renam) [reply kotlin/src/commonMain/kotlin/com/adsbynimbus/openrtb/request/Asset.kt]: I removed it

**Review by JasonSznol — APPROVED** [review]

**Pull request merged**